spa/src/hooks/useConnectMailchimp.tsx

```diff
@@ -81,13 +81,14 @@ export default function useConnectMailchimp(): UseConnectMailchimpResult {
 
   // Only require audience selection if there is no list selected, and if the
   // audience lists is populated (mailchimp connection successfully started).
-
   const requiresAudienceSelection =
     !mailchimpData?.mailchimp_list_id && (mailchimpData?.available_mailchimp_email_lists?.length ?? 0) > 0;
 
+  // To know when mailchimp has successfully finalized the entire connection process,
+  // we need to know when the user has just selected an audience.
   const justConnectedToMailchimp = useMemo(
-    () => !!(mailchimpData?.mailchimp_integration_connected && prevMailchimpConnection === false),
-    [mailchimpData?.mailchimp_integration_connected, prevMailchimpConnection]
+    () => !!mailchimpData?.mailchimp_list_id && !prevMailchimpAudienceId,
```

Comment:
Let's be explicit here since `mailchimp_list_id` is no longer a boolean.

```suggestion
    () => (typeof mailChimpData?.mailchimp_list_id === 'number' && prevMailchimpAudienceId === null),
```

Reply:
(or 'string', probably)